edb/server/http.py

```diff
@@ -0,0 +1,191 @@
+
+    async def post(
+        self,
+        path: str,
+        *,
+        headers: dict[str, str] | None = None,
+        data: bytes | str | dict[str, str] | None = None,
+        json: Any | None = None,
+    ) -> Response:
+        if json is not None:
+            data = json_lib.dumps(json).encode('utf-8')
+            headers = headers or {}
+            headers['Content-Type'] = 'application/json'
+        elif isinstance(data, str):
+            data = data.encode('utf-8')
+        elif isinstance(data, dict):
+            data = urllib.parse.urlencode(data).encode('utf-8')
+            headers = headers or {}
+            headers['Content-Type'] = 'application/x-www-form-urlencoded'
+
+        headers_list = [(k, v) for k, v in headers.items()] if headers else None
+        result = await self.request(
+            method="POST", url=path, content=data, headers=headers_list
+        )
+        return Response.from_tuple(result)
+
+    async def _boot(self, loop: asyncio.AbstractEventLoop) -> None:
+        logger.info("Python-side HTTP client booted")
+        reader = asyncio.StreamReader(loop=loop)
+        reader_protocol = asyncio.StreamReaderProtocol(reader)
+        fd = os.fdopen(self._client._fd, 'rb')
+        transport, _ = await loop.connect_read_pipe(lambda: reader_protocol, fd)
+        try:
+            while len(await reader.read(1)) == 1:
+                if not self._client or not self._task:
+                    break
+                if self._skip_reads > 0:
+                    self._skip_reads -= 1
+                    continue
+                msg = self._client._read()
+                if not msg:
+                    break
+                self._process_message(msg)
+        finally:
+            transport.close()
+
+    def _process_message(self, msg):
+        msg_type, id, data = msg
+
+        if id in self._requests:
+            if msg_type == 1:
+                self._requests[id].set_result(data)
+            elif msg_type == 0:
+                self._requests[id].set_exception(Exception(data))
+
+
+class CaseInsensitiveDict(dict):
+    def __init__(self, data: Optional[list[Tuple[str, str]]] = None):
+        super().__init__()
+        if data:
+            for k, v in data:
+                self[k.lower()] = v
+
+    def __setitem__(self, key: str, value: str):
+        super().__setitem__(key.lower(), value)
+
+    def __getitem__(self, key: str):
+        return super().__getitem__(key.lower())
+
+    def get(self, key: str, default=None):
+        return super().get(key.lower(), default)
+
+    def update(self, *args, **kwargs: str) -> None:
+        if args:
+            data = args[0]
+            if isinstance(data, Mapping):
+                for key, value in data.items():
+                    self[key] = value
+            else:
+                for key, value in data:
+                    self[key] = value
+        for key, value in kwargs.items():
+            self[key] = value
```

Comment:
Is it really worth doing this?

Reply:
The getter is definitely necessary to make sure the older code works, but the setter is more for ensuring we don't hit any landmines with code expecting headers to work as it does in httpx. I can take another pass at this once I finish up the AI code and figure out exactly what level of compat is necessary.